### DISCUSSION TITLE
is dispatcher routing not work for udp cone nat mode?

### DISCUSSION THREAD

**Discussion by rurirei** [v2fly/v2ray-core#2408] [Q&A] [answered]
as common/protocol/udp/packetaddr introduces the cone nat to let a transport.link loads requests with mutilple destinations, thought app.dispatcher matches routing rules once with the first request, when initials one link only, causes the no matching rules with the destinations comes later. what i'm missing? thanks @xiaokangwang

**Comment by negaroo** [comment]
#2470

**Reply by rurirei** [reply]
not related.

**Comment by AkinoKaede** [comment, marked as answer]
Yes, as a connection may have multiple targets, which cannot be well matched by routing rules.
